model/NewSqlResultStorage.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2014-2019 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ */
+
+namespace oat\taoOutcomeRds\model;
+
+use Doctrine\DBAL\Schema\Schema;
+use taoResultServer_models_classes_Variable as Variable;
+
+/**
+ * Implements tao results storage for NewSql
+ */
+class NewSqlResultStorage extends AbstractRdsResultStorage
+{
+    const CREATED_AT = 'created_at';
+
+    /**
+     * @inheritDoc
+     */
+    protected function getVariablesSortingField()
+    {
+        return self::CREATED_AT;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function prepareVariableDataForSchema($deliveryResultIdentifier, $test, Variable $variable, $callId)
+    {
+        $serializedVariable = $this->serializeVariableValue($variable);
+
+        $persistence = $this->getPersistence();
+
+        return [
+            self::VARIABLES_TABLE_ID => $persistence->getUniquePrimaryKey(),
+            self::VARIABLES_FK_COLUMN => $deliveryResultIdentifier,
+            self::TEST_COLUMN => $test,
+            self::VARIABLE_IDENTIFIER => $variable->getIdentifier(),
+            self::VARIABLE_VALUE => $serializedVariable,
+            self::VARIABLE_HASH => $deliveryResultIdentifier . md5($deliveryResultIdentifier . $serializedVariable . $callId),
+            self::CREATED_AT => $this->microTimeToMicroSeconds($variable->getEpoch(), $persistence->getPlatform()->getDateTimeFormatString()),
```

Comment:
can use the new getCreationTime() instead of Epoch on variable

Reply:
Where does it sit? Something newly merged?

Reply:
https://github.com/oat-sa/tao-core/blame/master/models/classes/lock/Lock.php#L39
done 5 years ago by @jbout the interface
but the implementation does the following
```
    /**
     * 
     * @author "Patrick Plichart, <patrick@taotesting.com>"
     */
    public function getCreationTime(){
        return $this->epoch;
    }
```
so i think it just a matter of preference

Reply:
@Mike, this is the same method name, but not the right object...